lms/services/vitalsource/service.py

```diff
@@ -7,9 +7,16 @@
 class VitalSourceService:
     """A high-level interface for dealing with VitalSource."""
 
-    def __init__(self, client: VitalSourceClient):
+    def __init__(self, client: VitalSourceClient, enabled):
+        self._enabled = enabled
         self.client = client
 
+    @property
+    def enabled(self) -> bool:
```

Comment:
Is this `enabled` or `file_picker_enabled`?

Reply:
Yeah, it's probably that in a local sense, but I think a missing piece is that this probably should be a new type of file source concept in it's entirety (rather than a service). It's not obvious to a casual observer, but this is 100% about serving up that type of content. There's no other type of VitalSourcey thing we do.

The same goes for JSTOR. In that world it's really whether "VitalSource" or "GoogleDrive" etc is enabled on this account. In that world, it's kind of about the whole concept being enabled or disabled, rather than just the picker. For example, if you pick a file, and then we break the settings, we should probably do something more intelligent to warn you we can't get the doc than we do right now, which would also use the enabled setting.

Long story short:

* One point of view is this is a generic service which does many things, one of which is turning on and off the file picker
* Another is this is _all_ about VitalSource as a source of documents and it's enabled or disabled as a whole - and _currently_ that's only used in one place (the file picker)

I think I come down on the latter side. I'd really like to have a crack at creating a new concept around document sources at some point which include things like this.